public/Aframe/loading-screens/scene-1/components.js

```diff
@@ -0,0 +1,114 @@
+AFRAME.registerComponent("collision-detect", {
+    init: function () {
+        this.el.addEventListener("collide", function (e) {
+            console.log("Player has collided with ", e.detail.body.el);
+            e.detail.target.el; // Original entity (playerEl).
+            e.detail.body.el; // Other entity, which playerEl touched.
+            e.detail.contact; // Stats about the collision (CANNON.ContactEquation).
+            e.detail.contact.ni; // Normal (direction) of the collision (CANNON.Vec3).
+        });
+    },
+});
+
+AFRAME.registerComponent("detect-button", {
+    init: function () {
+        var data = this.data;
+        var el = this.el;
+
+        el.addEventListener("mouseenter", function () {
+            el.setAttribute(
+                "animation__button-enter",
+                "property: scale; to: 1.1 1.1 1.1; dur: 500; dir:alternate; startEvents: mouseenter"
+            );
+        });
+
+        el.addEventListener("mouseleave", function () {
+            el.setAttribute(
+                "animation__button-leave",
+                "property: scale; to: 1 1 1; startEvents: mouseleave"
+            );
+        });
+
+        el.addEventListener("mousedown", function () {
+            el.setAttribute(
+                "animation__button-pressed",
+                "property: object3D.position.y; to: -5.5; dir: alternate; dur: 50; startEvents: mousedown"
+            );
+
+            el.emit("buttondown", false);
+        });
+
+        el.addEventListener("mouseup", function () {
+            el.setAttribute(
+                "animation__button-released",
+                "property: object3D.position.y; to: -5; dir: alternate; dur: 100; startEvents: mouseup"
+            );
+        });
+    },
+});
+
+function addBall() {
+    let container = document.querySelector("#container");
+    let colors = [
+        "red",
+        "orange",
+        "yellow",
+        "green",
+        "blue",
+        "purple",
+        "hotpink",
+    ];
+
+    let x = Math.random() * 10 - 5;
+    let y = Math.random() * 5 + 2;
+    let z = Math.random() * -10;
+
+    container.innerHTML += `<a-sphere collision-detect id="ball" click-drag dynamic-body position="${x} ${y} ${z}" radius="0.5" color="${
+        colors[Math.floor(Math.random() * colors.length)]
+    }" mass="0.5"></a-sphere>`;
+}
+
+AFRAME.registerComponent("add-ball", {
+    update: function () {
+        var data = this.data;
+        var el = this.el;
+
+        var curBalls = document.querySelectorAll("#ball");
+
+        for (let i = 0; i < curBalls.length; i++) {
+            let container = document.querySelector("#container");
+            container.removeChild(curBalls[i]);
+        }
+
+        el.addEventListener("buttondown", function () {
```

Comment:
I was reading [this document](https://aframe.io/docs/1.2.0/introduction/writing-a-component.html#handling-property-updates) today and removing existing listeners before adding a new one is something that may be impacting performance for `update` events. I am not entirely sure though so I might be wrong, I have not gotten a chance to test it out myself yet

Reply:
Took a quick glance and so far looks promising. Not sure how I didn't come across this as I remember browsing through this section. Thanks for the notice